### PR TITLE
SQL Project - Fixing the telemetry property docker image name

### LINKED ISSUE #19683: SQL Project - Docker Base Image telemetry property flagged as user path
- Azure Data Studio Version:
- OS Version:

When publishing a SQL project to docker container, we add a telemetry including the image base used for the docker container. The image name is flagged as user path because it has "/" in the name "repo/imagename:tag"

### PR DESCRIPTION
**Pull request by llali**
This PR fixes # 19683

### CODE REVIEW

**Review by kisantia — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 1]
```diff
@@ -319,9 +319,12 @@ export class ProjectsController {
 	 */
 	public async publishToDockerContainer(context: Project | dataworkspace.WorkspaceTreeItem, deployProfile: ILocalDbDeployProfile): Promise<void> {
 		const project: Project = this.getProjectFromContext(context);
+		// Removing the path separator from the image base name to be able to add that in the telemetry. With the separator the name is flagged as user path which is not true
+		// We only need to know the image base parts so it's ok to use a different separator when adding to telemetry
+		const dockerImageNameForTelemetry = deployProfile.localDbSetting?.dockerBaseImage ? deployProfile.localDbSetting.dockerBaseImage.replace(/\//gi, '_') : '';
```
Comment (Charles-Gagnon) [inline extensions/sql-database-projects/src/controllers/projectController.ts:324]: fyi you can simplify this a bit more 

`= deployProfile.localDbSetting?.dockerBaseImage?.replace(/\//gi, '_') ?? ''`

Comment (llali) [reply extensions/sql-database-projects/src/controllers/projectController.ts:324]: for my understanding, deployProfile.localDbSetting?.dockerBaseImage? won't return undefined potentially? causing undefined error  for .replace?

Comment (Charles-Gagnon) [reply extensions/sql-database-projects/src/controllers/projectController.ts:324]: I'm not sure I understand, if `deployProfile.localDbSetting?.dockerBaseImage` is undefined then it won't call replace at all (it short circuits and will evaluate to undefined, which the `?? ''` then ensures is set as an empty string

**Comment by kisantia** [comment]
the "fixes" in the description will close the issue after this PR gets merged - please make sure the issue stays open until it gets ported to the release branch.

File: extensions/sql-database-projects/src/controllers/projectController.ts (hunk 1 above)
Comment (llali) [reply extensions/sql-database-projects/src/controllers/projectController.ts:324]: hmm ok I didn't know that. thanks.

Comment (Charles-Gagnon) [reply extensions/sql-database-projects/src/controllers/projectController.ts:324]: Example : https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCm0DyAjAVgRhgXhgbwCgZiYAzAS3gBsATDALnyJJYgFcULqbGAiACWpUQAegDqIAE60AhL2bEAvgWUFQkWAmToATDiYsutBgZYl2nSrUZswNeBTDwaCmMtXrocRFFRoAzPqEhlZ0NnYO5E4uHuAQIFTwAHTCAOYAFFq+6BhJRnQA-EkW+UWS8AAOVACGwPDpAOQiDQA0MA3tAJQwBQXtAKoRjs4yDZ1qcQnJaZk+fjp5oRhFJaFllTV1jc1tHWM9fQ2D9sM0o+OeUykgGVl+-ovcy8UcpUnlVbX1Ta3tXQcDIZREZjIA

**Pull request merged**